include/ddc/kernels/splines/constant_extrapolation_rule.hpp

```diff
@@ -0,0 +1,112 @@
+#pragma once
+
+#include "view.hpp"
+
+namespace ddc {
+
+template <class DimI, class... Dim>
+struct ConstantExtrapolationRule
+{
+};
+
+template <class DimI>
+struct ConstantExtrapolationRule<DimI>
+{
+private:
+    ddc::Coordinate<DimI> m_eval_pos;
+
+public:
+    explicit ConstantExtrapolationRule(ddc::Coordinate<DimI> eval_pos) : m_eval_pos(eval_pos) {}
+
+    template <class CoordType, class BSplines, class Layout, class MemorySpace>
+    KOKKOS_FUNCTION double operator()(
+            CoordType,
+            ddc::ChunkSpan<double const, ddc::DiscreteDomain<BSplines>, Layout, MemorySpace> const
+                    spline_coef) const
+    {
+        std::array<double, BSplines::degree() + 1> vals;
+
+        ddc::DiscreteElement<BSplines> idx
+                = ddc::discrete_space<BSplines>().eval_basis(vals, m_eval_pos);
+
+        double y = 0.0;
+        for (std::size_t i = 0; i < BSplines::degree() + 1; ++i) {
+            y += spline_coef(idx + i) * vals[i];
+        }
+        return y;
+    }
+};
+
+template <class DimI, class DimNI>
+struct ConstantExtrapolationRule<DimI, DimNI>
+{
+private:
+    ddc::Coordinate<DimI> m_eval_pos;
+    ddc::Coordinate<DimNI> m_eval_pos_not_interest_min;
+    ddc::Coordinate<DimNI> m_eval_pos_not_interest_max;
+
+public:
+    explicit ConstantExtrapolationRule(
+            ddc::Coordinate<DimI> eval_pos,
+            ddc::Coordinate<DimNI> eval_pos_not_interest_min,
+            ddc::Coordinate<DimNI> eval_pos_not_interest_max)
+        : m_eval_pos(eval_pos)
+        , m_eval_pos_not_interest_min(eval_pos_not_interest_min)
+        , m_eval_pos_not_interest_max(eval_pos_not_interest_max)
+    {
+    }
+
+    explicit ConstantExtrapolationRule(ddc::Coordinate<DimI> eval_pos)
+        : m_eval_pos(eval_pos)
+        , m_eval_pos_not_interest_min(0.)
+        , m_eval_pos_not_interest_max(0.)
+    {
+        static_assert(
+                DimNI::PERIODIC,
+                "Single-argument constructor of ConstantExtrapolationRule is available only for "
+                "periodic non-interest dimension");
+    }
+
+    template <class CoordType, class BSplines1, class BSplines2, class Layout, class MemorySpace>
+    KOKKOS_FUNCTION double operator()(
+            CoordType coord_extrap,
```

Comment:
Can you add constraints about `CoordType` ?

Reply:
something like:
`in_tag_v<DimI,ToTypeSeq<CoordType>> && in_tag_v<DimNI,ToTypeSeq<CoordType>>` ?

Reply:
Yes if it is the condition to make it work.

Reply:
Actually not. I think it can compile with `DimI` not in the CoordType, but it would be a bit weird and not an extrapolation so I guess it is simpler to static_assert it too